library/src/main/java/com/novoda/downloadmanager/Batch.java

```diff
@@ -68,15 +68,22 @@ public String toString() {
 
         private final DownloadBatchId downloadBatchId;
         private final String title;
-        private final List<String> fileUrls = new ArrayList<>();
+        private final Map<DownloadFileId, String> fileUrls = new HashMap<>();
 
         public Builder(DownloadBatchId downloadBatchId, String title) {
             this.downloadBatchId = downloadBatchId;
             this.title = title;
         }
 
         public Builder addFile(String fileUrl) {
-            fileUrls.add(fileUrl);
+            String id = downloadBatchId.stringValue() + fileUrl;
+            DownloadFileId downloadFileId = DownloadFileIdCreator.createFrom(id);
```

Comment:
If the client does not specify a custom identifier then one is created from the `downloadBatch` and the `fileUrl` as these should be unique.